ls1intum/Artemis
# `Lectures`: Show slide numbers to be removed in automatic unit processing

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
Solves  Issue #6896 

### Description
When adding attachment units to a lecture via automatic lecture processing, users can remove slides with certain keywords (e.g. "Break"). This PR adds the functionality that when the user searches for keywords, the slides that will be removed get displayed.

### Steps for Testing
1. Log in as an Editor/Instructor
2. Go to any course -> manage -> lectures
3. Create a new lecture or edit an existing one
4. Tick the checkbox _automatic unit processing_
5. Upload a .pdf file (Here is an [Example.pdf](https://github.com/ls1intum/Artemis/files/12883825/Example.pdf))
6. Click on _Process units_
7. (Optional) split the file into more than the units that already exist
8. Enter keyphrases, seperated by commas (e.g. "Break, Example Solution")
9. Observe that the textbox above gets updated with the slide numbers that get removed
10. Observe that empty keyphrases (e.g. "  ", "ValidKeyphrase,    ") don't get evaluated (For the latter "Valid Keyphrase gets evaluated but the empty keyphrase after it will not)
11. Click on save and verify that the generated units are correct

### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [attachment-units.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5563/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture-unit/lecture-unit-management/attachment-units/attachment-units.component.ts.html) | 90.56% | ✅  |
| [attachmentUnit.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5563/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture-unit/lecture-unit-management/attachmentUnit.service.ts.html) | 100% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [LectureUnitProcessingService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5563/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/LectureUnitProcessingService.html) | 93% | ✅  |
| [AttachmentUnitResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5563/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.lecture/AttachmentUnitResource.html) | 92% | ✅  |


### Screenshots
![screen1](https://github.com/ls1intum/Artemis/assets/118574504/a776cfa1-6f01-447b-bc42-c568b017d7c2)

![screen3](https://github.com/ls1intum/Artemis/assets/118574504/526a10b6-6edb-40f8-8760-dd6d866574c4)
